sqlglot/expressions.py

```diff
@@ -1632,6 +1632,7 @@ class AlterColumn(Expression):
         "default": False,
         "drop": False,
         "comment": False,
+        "expression": False
```

Comment:
Let's rename this to `not_null` so it's a bit clearer what it stands for.

Reply:
I will use `allow_null` instead as it seems to be more consistent with the `NotNullColumnConstraints` arg_type.